engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -950,112 +972,467 @@ static dmGameSystemDDF::Playback GameObjectPlaybackToDDFPlayback(dmGameObject::P
+            luaL_checktype(L, -1, LUA_TTABLE);
+            int animation_index = luaL_checkinteger(L, -2);
+
+            // Note: checkstring can change the lua stack, so we use isstring instead
+            lua_getfield(L, -1, "id");
+            if (!lua_isstring(L, -1))
+            {
+                luaL_error(L, "Invalid 'id' in animations table at index [%d], either missing or wrong type", num_animations + 1);
+            }
+            lua_pop(L, 1);
+
+            // Required fields
+            CheckFieldValue<int>(L, -1, "width");
+            CheckFieldValue<int>(L, -1, "height");
+            int frame_start = CheckFieldValue<int>(L, -1, "frame_start");
+            int frame_end   = CheckFieldValue<int>(L, -1, "frame_end");
+
+            // Non-required fields
+            CheckFieldValue<int>(L,  -1, "playback", 0);
+            CheckFieldValue<int>(L,  -1, "fps", 0);
+            CheckFieldValue<bool>(L, -1, "flip_vertical", false );
+            CheckFieldValue<bool>(L, -1, "flip_horizontal", false );
+
+            // Validate frame indices
+            int frame_interval = frame_end - frame_start;
+            if (frame_start < 1 || frame_start > (num_geometries+1)) // +1 for lua indexing
+            {
+                luaL_error(L, "Invalid frame_start in animation [%d], index %d is outside of geometry bounds 0..%d",
+                        animation_index, frame_start, num_geometries);
+            }
+
+            if (frame_end < 1 || frame_end > (num_geometries+1)) // +1 for lua indexing
+            {
+                luaL_error(L, "Invalid frame_end in animation [%d], index %d is outside of geometry bounds 0..%d",
+                    animation_index, frame_end, num_geometries);
+            }
+
+            if (frame_interval <= 0)
+            {
+                luaL_error(L, "Invalid frame interval in animation [%d], start - end = %d", animation_index, frame_interval);
+            }
+
+            lua_pop(L, 1);
+
+            num_animations++;
         }
-        lua_pop(L, 1);
+    }
+
+    lua_pop(L, 1);
+
+    *num_animations_out = num_animations;
+    *num_geometries_out = num_geometries;
+
+    if (num_geometries == 0)
+    {
+        luaL_error(L, "Atlas requires at least one entry in the 'geometries' table");
+    }
+    if (num_animations == 0)
+    {
+        luaL_error(L, "Atlas requires at least one entry in the 'animations' table");
+    }
+
+    assert(lua_gettop(L) == top);
+}
+
+// Creates a texture set from the lua stack, it is expected that the argument
+// table is on top of the stack and that all fields have valid data
+static void MakeTextureSetFromLua(lua_State* L, const char* texture_path, dmGraphics::HTexture texture, uint32_t num_geometries, uint8_t num_animations, dmGameSystemDDF::TextureSet* texture_set_ddf)
+{
+    int top = lua_gettop(L);
+
+    texture_set_ddf->m_Texture = texture_path;
```

Comment:
See note above regarding hash of texture resource path.

Apart from creating resources when loading from disc, what other uses does this string have?
Possibly this could be 0, and instead we set a hash value?